src/main/java/gregtech/loaders/recipe/RecyclingRecipes.java

```diff
@@ -1,111 +1,402 @@
-                .EUt(8 * voltageMultiplier);
-        inputSupplier.accept(maceratorRecipeBuilder);
-        maceratorRecipeBuilder.buildAndRegister();
-
-        if (material.hasFluid()) {
-            RecipeBuilder<?> fluidExtractorRecipeBuilder = RecipeMaps.EXTRACTOR_RECIPES.recipeBuilder()
-                    .fluidOutputs(material.getFluid((int) (firstStack.amount * L / M)))
-                    .duration((int) Math.max(1L, firstStack.amount * 80 / M))
-                    .EUt(32 * voltageMultiplier);
-            inputSupplier.accept(fluidExtractorRecipeBuilder);
-            fluidExtractorRecipeBuilder.buildAndRegister();
-        }
-
-        if (!ignoreArcSmelting) {
-            List<ItemStack> resultList = materials.stream().map(RecyclingRecipes::getArcSmeltingResult).collect(Collectors.toList());
-            resultList.removeIf(ItemStack::isEmpty);
-            if (resultList.isEmpty()) return;
-            RecipeBuilder<?> arcFurnaceRecipeBuilder = RecipeMaps.ARC_FURNACE_RECIPES.recipeBuilder()
-                    .outputs(resultList)
-                    .duration((int) Math.max(1L, firstStack.amount * 60 / M))
-                    .EUt(30 * voltageMultiplier);
-            inputSupplier.accept(arcFurnaceRecipeBuilder);
-            arcFurnaceRecipeBuilder.buildAndRegister();
+
+        // Calculate the voltage multiplier based on if a Material has a Blast Property
+        int voltageMultiplier = calculateVoltageMultiplier(components);
+
+        registerMaceratorRecycling(input, components, voltageMultiplier);
+        registerExtractorRecycling(input, components, voltageMultiplier);
+        if (ignoreArcSmelting) return;
+
+        //do not apply arc smelting for gems, solid materials and dust materials
+        //only generate recipes for ingot materials
+        if (components.size() == 1) {
+            Material m = components.get(0).material;
+            if (!(m.hasProperty(PropertyKey.INGOT) && m.getProperty(PropertyKey.INGOT).getArcSmeltInto() != m)) {
+                return;
+            }
         }
+        registerArcRecycling(input, components, voltageMultiplier);
     }
 
-    private static ItemStack getArcSmeltingResult(MaterialStack materialStack) {
+    private static void registerMaceratorRecycling(ItemStack input, List<MaterialStack> materials, int multiplier) {
+
+        // Finalize the output list.
+        List<ItemStack> outputs = finalizeOutputs(
+                materials,
+                RecipeMaps.MACERATOR_RECIPES.getMaxOutputs(),
+                OreDictUnifier::getDust
+        );
+
+        // Exit if no valid Materials exist for this recycling Recipe.
+        if (outputs.size() == 0) return;
+
+        // Build the final Recipe.
+        RecipeMaps.MACERATOR_RECIPES.recipeBuilder()
+                .inputs(input.copy())
+                .outputs(outputs)
+                .duration(calculateDuration(materials))
+                .EUt(4 * multiplier)
```

Comment:
Macerator default EU/t has been changed to 2 instead of 4, as that is what it is in 5u
(also, extractor and arc EU/t should probably be changed to VA[LV])